### PR TITLE
fix missing required attributes

### PR DESCRIPTION
**Pull request by M4hd1BD**

### CODE REVIEW

File: templates/account/registration.php [hunk 1]
```diff
@@ -49,19 +49,21 @@
...
+							<input id="email" class="directorist-form-element" type="text" name="email" value="<?php echo isset( $_REQUEST['email'] ) ? esc_attr( sanitize_email( wp_unslash( $_REQUEST['email'] ) ) ) : ''; ?>" required>
 						</div>
 						<?php if ( ! empty( $display_password_reg ) ) { ?>
 							<div class="directorist-form-group directorist-mb-15">
 								<label for="password"><?php
 									echo esc_html( $password );
 									echo ( ! empty( $require_password ) ? '<strong class="directorist-form-required">*</strong>' : '' );
 								?></label>
-								<input id="password" class="directorist-form-element" type="password" name="password" value="">
+								<input id="password" class="directorist-form-element" type="password" name="password" value="" <?php
```
Comment (kowsar89) [inline templates/account/registration.php:64]: Keep it in single line, like before

File: templates/account/registration.php [hunk 2]
```diff
@@ -116,13 +126,13 @@
...
+								<input id="privacy_policy" type="checkbox" name="privacy_policy" <?php echo( ( isset( $privacy_policy ) && 'on' === $privacy_policy ) ? 'checked="checked"' : '' ); ?> required>
 								<label for="privacy_policy" class="directorist-checkbox__label"><?php echo esc_html( $privacy_label ); ?> <a style="color: red" target="_blank" href="<?php echo esc_url( $privacy_page_link ); ?>"><?php echo esc_html( $privacy_label_link ); ?></a> <span class="directorist-form-required">*</span></label>
 							</div>
 						<?php } ?>
 						<?php if ( ! empty( get_directorist_option('regi_terms_condition', 1 ) ) ) { ?>
 							<div class="atbd_term_and_condition_area directory_regi_btn directorist-checkbox directorist-mb-15">
-								<input id="listing_t" type="checkbox" name="t_c_check" <?php echo( ( isset( $t_c_check ) && 'on' === $t_c_check ) ? 'checked="checked"' : '' ); ?>>
+								<input id="listing_t" type="checkbox" name="t_c_check" <?php echo( ( isset( $t_c_check ) && 'on' === $t_c_check ) ? 'checked="checked"' : '' ); ?> required>
```
Comment (kowsar89) [inline templates/account/registration.php:135]: doesn't work

File: templates/account/registration.php [hunk 3]
```diff
@@ -116,13 +126,13 @@
 
 						<?php if ( ! empty( get_directorist_option( 'registration_privacy', 1 ) ) ) { ?>
 							<div class="atbd_privacy_policy_area directory_regi_btn directorist-checkbox directorist-mb-15">
-								<input id="privacy_policy" type="checkbox" name="privacy_policy" <?php echo( ( isset( $privacy_policy ) && 'on' === $privacy_policy ) ? 'checked="checked"' : '' ); ?>>
+								<input id="privacy_policy" type="checkbox" name="privacy_policy" <?php echo( ( isset( $privacy_policy ) && 'on' === $privacy_policy ) ? 'checked="checked"' : '' ); ?> required>
```
Comment (kowsar89) [inline templates/account/registration.php:129]: doesn't work

File: templates/account/registration.php [hunk 4]
```diff
@@ -70,7 +72,9 @@
 								echo esc_html( $first_name );
 								echo ( ! empty( $require_fname ) ? '<strong class="directorist-form-required">*</strong>' : '' );
 							?></label>
-							<input id="fname" class="directorist-form-element" type="text" name="fname" value="<?php echo isset( $_REQUEST['fname']) ? esc_attr( sanitize_text_field( wp_unslash( $_REQUEST['fname'] ) ) ) : ''; ?>">
+							<input id="fname" class="directorist-form-element" type="text" name="fname" value="<?php echo isset( $_REQUEST['fname']) ? esc_attr( sanitize_text_field( wp_unslash( $_REQUEST['fname'] ) ) ) : ''; ?>" <?php
```
Comment (kowsar89) [inline templates/account/registration.php:75]: Keep it in single line, like before

File: templates/account/registration.php [hunk 5]
```diff
@@ -79,7 +83,9 @@
 								echo esc_html( $last_name );
 								echo ( ! empty( $require_lname ) ? '<strong class="directorist-form-required">*</strong>' : '' );
 							?></label>
-							<input class="directorist-form-element" id="lname" type="text" name="lname" value="<?php echo isset( $_REQUEST['lname']) ? esc_attr( sanitize_text_field( wp_unslash( $_REQUEST['lname'] ) ) ) : ''; ?>">
+							<input class="directorist-form-element" id="lname" type="text" name="lname" value="<?php echo isset( $_REQUEST['lname']) ? esc_attr( sanitize_text_field( wp_unslash( $_REQUEST['lname'] ) ) ) : ''; ?>" <?php
```
Comment (kowsar89) [inline templates/account/registration.php:86]: Keep it in single line, like before

File: templates/account/registration.php [hunk 6]
```diff
@@ -88,7 +94,9 @@
 									echo esc_html( $website );
 									echo ( ! empty( $require_website ) ? '<strong class="directorist-form-required">*</strong>' : '' );
 								?></label>
-								<input id="website" class="directorist-form-element" type="text" name="website" value="<?php echo isset( $_REQUEST['website']) ? esc_url( sanitize_text_field( wp_unslash( $_REQUEST['website'] ) ) ) : ''; ?>">
+								<input id="website" class="directorist-form-element" type="text" name="website" value="<?php echo isset( $_REQUEST['website']) ? esc_url( sanitize_text_field( wp_unslash( $_REQUEST['website'] ) ) ) : ''; ?>" <?php
```
Comment (kowsar89) [inline templates/account/registration.php:97]: Keep it in single line, like before

File: templates/account/registration.php [hunk 7]
```diff
@@ -97,7 +105,9 @@
 								echo esc_html( $bio );
 								echo ( ! empty( $require_bio ) ? '<strong class="directorist-form-required">*</strong>' : '' );
 							?></label>
-							<textarea id="bio" class="directorist-form-element" name="bio" rows="10"><?php echo isset( $_REQUEST['bio']) ? esc_textarea( sanitize_text_field( wp_unslash( $_REQUEST['bio'] ) ) ) : ''; ?></textarea>
+							<textarea id="bio" class="directorist-form-element" name="bio" rows="10" <?php
```
Comment (kowsar89) [inline templates/account/registration.php:108]: Keep it in single line, like before

**Review by kowsar89 — CHANGES_REQUESTED** [review]

File: templates/account/registration.php (hunk 1 above)
Comment (M4hd1BD) [reply templates/account/registration.php:64]: Done

File: templates/account/registration.php (hunk 2 above)
Comment (M4hd1BD) [reply templates/account/registration.php:135]: Got rid of this. The Checkbox is custom here, this one isn't focusable, need JS to throw error here.

File: templates/account/registration.php (hunk 3 above)
Comment (M4hd1BD) [reply templates/account/registration.php:129]: Got rid of this. The Checkbox is custom here, this one isn't focusable, need JS to throw error here.

File: templates/account/registration.php (hunk 4 above)
Comment (M4hd1BD) [reply templates/account/registration.php:75]: Done,

File: templates/account/registration.php (hunk 6 above)
Comment (M4hd1BD) [reply templates/account/registration.php:97]: Done.

File: templates/account/registration.php (hunk 7 above)
Comment (M4hd1BD) [reply templates/account/registration.php:108]: Done.

File: templates/account/registration.php [hunk 8]
```diff
@@ -97,7 +97,7 @@
 								echo esc_html( $bio );
 								echo ( ! empty( $require_bio ) ? '<strong class="directorist-form-required">*</strong>' : '' );
 							?></label>
-							<textarea id="bio" class="directorist-form-element" name="bio" rows="10"><?php echo isset( $_REQUEST['bio']) ? esc_textarea( sanitize_text_field( wp_unslash( $_REQUEST['bio'] ) ) ) : ''; ?></textarea>
+							<textarea id="bio" class="directorist-form-element" name="bio" rows="10" <?php echo ( ! empty( $require_bio ) ? 'required' : '' ); ?>> <?php echo isset( $_REQUEST['bio']) ? esc_textarea( sanitize_text_field( wp_unslash( $_REQUEST['bio'] ) ) ) : ''; ?></textarea>
```
Comment (kowsar89) [inline templates/account/registration.php:100]: 'required' doesn't work

**Review by kowsar89 — CHANGES_REQUESTED** [review]

File: templates/account/registration.php (hunk 8 above)
Comment (M4hd1BD) [reply templates/account/registration.php]: Fixed.

**Review by kowsar89 — APPROVED** [review]
Looks good

**Pull request merged**